stdlib/public/core/TypeDiscovery.swift

```diff
@@ -0,0 +1,57 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+@_silgen_name("swift_enumerateConformingTypesFromImage")
+private func _swift_enumerateTypes(
+  fromImageAt imageAddress: UnsafeRawPointer?,
```

Comment:
`imageAddress` probably actually wants to be a `dlopen()` handle. On Apple platforms it's tricky getting from one to the other, and user code most likely has the latter.

Reply:
That's at least sometimes true. `#dsohandle` is the Mach header on Apple platforms and we should definitely support that, but you're right that the most obvious other input would be a `dlopen()` handle.

I suppose we can just say "pass the `__dso_handle` symbol in that case" which is easy to get, though I'm not sure if it's exported if nobody uses it in-source. I'll ping the dyld maintainers for advice off-GitHub.

Reply:
`#dsohandle` should give you the equivalent address on Windows (`__ImageBase`) and the same address as `dlopen` on Linux.